internal/p2p/peermanager.go

```diff
@@ -1175,9 +1201,47 @@ func (s *peerStore) Ranked() []*peerInfo {
 		s.ranked = append(s.ranked, peer)
 	}
 	sort.Slice(s.ranked, func(i, j int) bool {
-		// FIXME: If necessary, consider precomputing scores before sorting,
-		// to reduce the number of Score() calls.
-		return s.ranked[i].Score() > s.ranked[j].Score()
+		// sort inactive peers after active peers
+		if s.ranked[i].Inactive && !s.ranked[j].Inactive {
+			return false
+		} else if !s.ranked[i].Inactive && s.ranked[j].Inactive {
+			return true
+		}
+
+		iLastDialed, iLastDialSuccess := s.ranked[i].LastDialed()
+		jLastDialed, jLastDialSuccess := s.ranked[j].LastDialed()
+
+		// sort peers who our most recent dialing attempt was
+		// successful ahead of peers with recent dialing
+		// failures
+		switch {
```

Comment:
Score already incorporates this logic by subtracting one for every failed dial. Peers who have previously failed will naturally be scored lower

Reply:
but it doesn't increase for successful dials, and "I just dialed you and succeeded after failing for a while" is a problem. 

I also think that given that the small range of possible scores (before this) we end up not really capturing (meaningfully) "this peer is real"

Reply:
Does it need to increase for successful dials when we reset the counter after being successful. That should be enough to sufficiently offset your score

Reply:
this is commented out and we can revisit it later.